elastic/kibana
# [Task Manager] Add caching to the task partitioning logic

Resolves https://github.com/elastic/kibana/issues/189119

## Summary

This PR adds a mechanism to keep the node's calculated partitions in cache for 10 seconds before calling the discovery service again and recalculating them.


### Checklist



### To verify
- Add the following to kibana.yml:
```
xpack.task_manager.claim_strategy: 'unsafe_mget'
```
- Verify that the cache is being updated every 10 seconds, you could test this by adding a log statement with the timestamp in the TaskPartitioner code that was updated
- Verify that on start the cache is updated on the initial claiming cycle
